File: ios/LivenessViewManager.m
//
//  LivenessViewManager.m
//  bringSwiftUIToReactNative2
//
//  Created by Chong Jhuang Hung on 06/11/2023.
//
#import <Foundation/Foundation.h>
#import <React/RCTUtils.h>
#import "React/RCTViewManager.h"
#import <UIKit/UIKit.h>
#import <React/RCTUIManager.h>
@interface RCT_EXTERN_MODULE(LivenessViewManager, RCTViewManager)
- (dispatch_queue_t)methodQueue
{
 return dispatch_queue_create("com.amazonaws.faceliveness.cameracapturequeue", DISPATCH_QUEUE_SERIAL);
}
@end

File: ios/LivenessViewManager.swift
//
//  LivenessViewManager.swift
//  bringSwiftUIToReactNative2
//
//  Created by Chong Jhuang Hung on 06/11/2023.
//

import Foundation
import UIKit
import SwiftUI

@objc(LivenessViewManager)
class LivenessViewManager: RCTViewManager {
  
  override var methodQueue: DispatchQueue! {
    return DispatchQueue(label: "com.amazonaws.faceliveness.cameracapturequeue")
  }
  
  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
  
  override func view() -> UIView! {
    let awsLivenessViewVC = UIHostingController(rootView: AWSFaceLivenessView())
    return awsLivenessViewVC.view
//    return AWSLivenessViewController().view
  }

  @objc(presentViewController)
  func presentViewController() {
    DispatchQueue.main.async {
//      let viewController = AWSLivenessViewController()
      let awsLivenessViewVC = UIHostingController(rootView: AWSFaceLivenessView())
      let rootViewController = RCTPresentedViewController()

//      rootViewController?.present(viewController, animated: true, completion: nil)
      rootViewController?.present(awsLivenessViewVC, animated: true, completion: nil)
    }
  }}
